js-core/homeworks/homework-6/src/main.js

```diff
@@ -0,0 +1,502 @@
+};
+
+/*
+ *
+ * TASK 1
+ * Посчитайте количество букв а, в передаваемом параметре
+ *
+ * */
+
+let randomString = 'aaa b a w c ';
+
+let user = {
+  name: 'Albina',
+};
+
+let javaScript = {
+  html: 'JavaScript',
+};
+
+function countLetterA(str) {
+  let strToArr = str.split('');
+  let aQuanlity = 0;
+  strToArr.forEach(function(elem){
+    if(elem == 'a'){
+      aQuanlity++;
+    }
+  });
+  return aQuanlity;
+}
+
+console.log(countLetterA(randomString)); // 4
+console.log(countLetterA(user.name + javaScript.html)); // 3
+
+// // ! ! ! ! ! ! ! ! ! ! !
+// /*
+//  *
+//  * TASK 2
+//  * Вам дано предложение, вам необходимо перевернуть каждое
+//  * слово в предложении в обратном порядке
+//  *
+//  * */
+
+function reverseEachWord(str) {
+  let arrToStr = str.split(' ');
+  let reversStr = arrToStr.map(function(elem) {
+    return elem.split('').reverse().join('')
+  });
+  return reversStr.join(' ');
+};
+
+
+console.log(reverseEachWord('You don\'t have to do anything special to begin using the DOM. Different browsers have different implementations of the DOM'));
+// uoY t'nod evah ot od gnihtyna laiceps ot nigeb gnisu eht .MOD tnereffiD sresworb evah tnereffid snoitatnemelpmi fo eht MOD
+
+console.log(reverseEachWord('The Document Object Model (DOM) is a programming interface for HTML and XML documents'));
+// ehT tnemucoD tcejbO ledoM )MOD( si a gnimmargorp ecafretni rof LMTH dna LMX stnemucod
+
+// /*
+//  * TASK 3
+//  * Добавьте в функцию reverseEachWord второй параметр,
+//  * данный параметр булево, если true - тогда предложение так же
+//  * переворачиваются в обратном порядке
+//  * */
+
+  function reverseEachWord(str, boolean){
+    let arrToStr = str.split(' ');
+    let reversToStr = arrToStr.map(function(elem){
+      return elem.split('').reverse().join('')
+    })
+    if(boolean === true){
+      return reversToStr.reverse().join(' ');
+    }
+    return reversToStr.join(' ');
+  };
+
+console.log(reverseEachWord('You don\'t have to do anything special to begin using the DOM. Different browsers have different implementations of the DOM', true));
+//MOD eht fo snoitatnemelpmi tnereffid evah sresworb tnereffiD .MOD eht gnisu nigeb ot laiceps gnihtyna od ot evah t'nod uoY
+console.log(reverseEachWord('The Document Object Model (DOM) is a programming interface for HTML and XML documents', true));
+// stnemucod LMX dna LMTH rof ecafretni gnimmargorp a si )MOD( ledoM tcejbO tnemucoD ehT
+
+console.log(reverseEachWord('Hi my Name is', false));
+// // iH ym emaN si
+
+// /* TASK 3
+//  * Посчитайте сколько одинаковых слов в предложении.
+//  * Функция должна возвращать объект у которого свойства
+//  * это повторяющиеся слово,
+//  * а значение это количество повторений
+//  * */
+
+// // Both - Java - and - Java - Script - is - programming - and - programming - OOPBased
+
+function wordCounter(sentence) {
+  let strToArr = sentence.split(' ');
+  let obj = strToArr.reduce(function(newValue, elem, index, arr) {
```

Comment:
you can solve that task with 1 loop. 

And yes **.reduce** fit very well for that task